### ISSUE TITLE
providers/aws: aws_autoscaling_group -/+ should wait till scaling group is removed before recreating

### ISSUE THREAD

**Issue by pmoust** [hashicorp/terraform#1105]
@catsby since `taint` command has been introduced I wanted to utilize it to iterate through some changes fast on our ASGs.

`Terraform v0.4.0-dev (23d90c0c02c10596eed79986e356b20bc6abb441)`

```
 ~/pph/pph-iac/peopleperhour ⮀ ⭠ master  ⮀ terraform taint aws_autoscaling_group.micro_staging
The resource aws_autoscaling_group.micro_staging in the module root has been marked as tainted!
 ~/pph/pph-iac/peopleperhour ⮀ ⭠ master ● ⮀ terraform plan -out /tmp/plan
<snipped>
-/+ aws_autoscaling_group.micro_staging
<snipped>
 ~/pph/pph-iac/peopleperhour ⮀ ⭠ master ● ⮀ terraform apply /tmp/plan
aws_autoscaling_group.micro_staging: Destroying...
aws_autoscaling_group.micro_staging: Creating... 
<snipped>
aws_autoscaling_group.micro_staging: Error: 1 error(s) occurred:

* Error creating Autoscaling Group: AutoScalingGroup by this name already exists - A group with the name micro_staging already exists and is pending delete.  Please wait until Autoscaling completes the deletion process before creating another group with the same name.
```
I 'd like TF to wait until the `aws_autoscaling_group` no longer exists before attempting to recreate it. It can take a while since instances need to be terminated first by the ASG scheduler.

**Comment by catsby** [comment]
@pmoust odd, the ASG resource is supposed to drain the instances before attempting to delete the ASG itself, so this _shouldn't_ be an issue. If you have a sample config around that would help, otherwise I'll try to reproduce this on my own. 
Thanks!

**Comment by catsby** [comment]
I've reproduced this, digging in more...

**Comment by catsby** [comment]
This is a timing problem, in that there's a brief window of time when the ASG has a status of "Delete in progress" after we issue the API call to delete it. We're not giving it enough time to complete the deletion before we try to re-create it. 

I'm looking into what we can do here. It's simple enough to poll for a successful deletion here before moving on, but that would likely block in every other scenario where it might not be necessary to wait.

**Comment by catsby** [comment]
@pmoust this should be fixed in #1353
